### ISSUE TITLE
SERVER ISSUE: Player progress no longer being saved

### ISSUE THREAD

**Issue by UnknownViper** [StrangeLoopGames/EcoIssues#4519]
Our current game has been corrupted in the sense that player progress is no longer being saved, meaning that a server restart will roll back all player progress (world changes unaffected) to a specific point (more on this later).

I initially noticed the issue when skill points were no longer being awarded. The skill points progress bar is just stuck and nobody is recieving skill points.

When people log in it says they were last online -33 seconds ago (usually a time around negative 32-38 seconds), likewise the amount of calories lost since last online can be in the negative.

As mentioned the character is rolled back to a specific point in time, meaning any items they've added to their inventory or equipped will cease to exist. This in turn also means that whatever you had at that point in time will be restored if used. If anything is placed in the world it will remain in the world, this issue seems to only affect saving the player state.

We were 8 or 9 days into the game, and we were on easy trajectory to win the game (I guess we made it a bit too easy on ourselves), so we wanted to check more of the cooking stuff out. Sadly we had removed wheat from the game (basically there was none to begin with), so I did a "/give cookingskillbook", learned it and then did "/repopulate Wheat". Now here's where I messed up. I confused repopulate with regenlayer. My assumpting is that repopulate messed everything up. Surprisingly the command also resulted in the meteor getting reset back to 30 days, as well as the server day progress (back to day 0).
My guess is that the character rollback is to the point of when the repopulate command happened (or was subsequently saved), at least it seems to match with where my character is spawned, as well as the state it's in.

Long story short I guess /repopulate has the risk of messing things up. I can supply the server game files if needed.

The game was started in 0.7.0.0 and progress with each update released. The repopulate command was issued in 0.7.1.0.

**Comment by UnknownViper** [comment]
I should add that I checked the error dumps, but there weren't anything related to this.

**Comment by MarjieAVolk** [comment]
Oh man... I bet you are right.  Since /repopulate resets the world time to 0, things might stop getting ticked because they are waiting for a specific world time in order to tick...

**Comment by MarjieAVolk** [comment]
We need to fix that command or disabled it @knexer

**Comment by knexer** [comment]
IMO the repopulate command should be removed.

**Comment by craig-jong** [comment]
The repopulate command has been removed.
